gcp-client/src/main/java/com/google/graphite/platforms/plugin/client/ComputeClient.java

```diff
@@ -0,0 +1,667 @@
+   * @throws InterruptedException If snapshot creation is interrupted.
+   */
+  public void createSnapshotSync(
+      final String projectId, final String zoneLink, final String instanceId, final long timeout)
+      throws IOException, InterruptedException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(zoneLink));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(instanceId));
+    Preconditions.checkArgument(timeout > 0);
+    String zoneName = nameFromSelfLink(zoneLink);
+    Instance instance;
+    try {
+      instance = compute.getInstance(projectId, zoneName, instanceId);
+    } catch (IOException ioe) {
+      LOGGER.log(Level.WARNING, "Error retrieving instance.", ioe);
+      throw ioe;
+    }
+
+    instance
+        .getDisks()
+        .parallelStream()
+        .forEach(
+            Errors.rethrow()
+                .wrap(
+                    disk -> {
+                      try {
+                        createSnapshotForDiskSync(
+                            projectId, zoneName, nameFromSelfLink(disk.getSource()), timeout);
+                      } catch (IOException ioe) {
+                        LOGGER.log(Level.WARNING, "Error in creating snapshot.", ioe);
+                        throw ioe;
+                      } catch (InterruptedException ie) {
+                        /* catching InterruptedException here because calling function also can
+                         * throw InterruptedException from trying to terminate node */
+                        LOGGER.log(Level.WARNING, "Interruption in creating snapshot.", ie);
+                        throw ie;
+                      }
+                    }));
+  }
+
+  /**
+   * Given a disk's name, create a snapshot for said disk. This method blocks until the operation
+   * complete.
+   *
+   * @param projectId The ID of the project where the disk resides.
+   * @param zoneName Zone of disk.
+   * @param diskName Name of disk to create a snapshot for.
+   * @param timeout The number of milliseconds to wait for snapshot creation.
+   * @throws IOException If an error occured in snapshot creation.
+   * @throws InterruptedException If snapshot creation is interrupted.
+   */
+  public Operation.Error createSnapshotForDiskSync(
+      final String projectId, final String zoneName, final String diskName, final long timeout)
+      throws IOException, InterruptedException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(zoneName));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(diskName));
+    Preconditions.checkArgument(timeout > 0);
+    Snapshot snapshot = new Snapshot();
+    snapshot.setName(diskName);
+
+    Operation op = compute.createDiskSnapshot(projectId, zoneName, diskName, snapshot);
+    // poll for result
```

Comment:
This comment is probably not needed as it seems clear what's going on.

Reply:
Done